### PR TITLE
Resurrect basic nninf checker

### PR DESCRIPTION
**Pull request by zcai1**

### CODE REVIEW

File: src/nninf/qual/NonNull.java [hunk 1]
```diff
@@ -1,38 +1,34 @@
 package nninf.qual;
 
-import org.checkerframework.framework.qual.DefaultFor;
-import org.checkerframework.framework.qual.LiteralKind;
-import org.checkerframework.framework.qual.QualifierForLiterals;
-import org.checkerframework.framework.qual.SubtypeOf;
-import org.checkerframework.framework.qual.TypeKind;
-import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedPrimitiveType;
+import org.checkerframework.framework.qual.*;
```
Comment (xingweitian) [inline src/nninf/qual/NonNull.java:3]: Please do not use a wild card with a Java import statement.

File: src/nninf/qual/Nullable.java [hunk 2]
```diff
@@ -1,8 +1,6 @@
 package nninf.qual;
 
-import org.checkerframework.framework.qual.LiteralKind;
-import org.checkerframework.framework.qual.QualifierForLiterals;
-import org.checkerframework.framework.qual.SubtypeOf;
+import org.checkerframework.framework.qual.*;
```
Comment (xingweitian) [inline src/nninf/qual/Nullable.java:3]: Same here.

File: src/nninf/qual/NonNull.java [hunk 3]
```diff
@@ -1,38 +1,34 @@
...
+                TypeKind.INT,
+                TypeKind.BOOLEAN,
+                TypeKind.CHAR,
+                TypeKind.DOUBLE,
+                TypeKind.FLOAT,
+                TypeKind.LONG,
+                TypeKind.SHORT,
+                TypeKind.BYTE
+        })
 public @interface NonNull {}
```
Comment (xingweitian) [inline src/nninf/qual/NonNull.java:34]: Could you please add a new line at here?

File: testdata/nninf/SimpleNninfTest2.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import nninf.qual.*;
```
Comment (xingweitian) [inline testdata/nninf/SimpleNninfTest2.java:1]: Same here.

File: testdata/nninf/SimpleNninfTest3.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import nninf.qual.*;
```
Comment (xingweitian) [inline testdata/nninf/SimpleNninfTest3.java:1]: Same here.

File: tests/checkers/inference/NninfTest.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    @Override
+    public Pair<String, List<String>> getSolverNameAndOptions() {
+        return Pair.<String, List<String>>of(MaxSat2TypeSolver.class.getCanonicalName(), new ArrayList<String>());
+    }
+
+    @Parameters
+    public static List<File> getTestFiles() {
+        List<File> testfiles = new ArrayList<>();//InferenceTestUtilities.findAllSystemTests();
```
Comment (xingweitian) [inline tests/checkers/inference/NninfTest.java:27]: Is it ok to remove the comment? What is this comment for?

File: tests/checkers/inference/NninfTest.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+public class NninfTest extends CFInferenceTest {
+
+    public NninfTest(File testFile) {
+        super(testFile, nninf.NninfChecker.class, "nninf",
+                "-Anomsgtext", "-d", "tests/build/outputdir");
+    }
+
+    @Override
+    public Pair<String, List<String>> getSolverNameAndOptions() {
+        return Pair.<String, List<String>>of(MaxSat2TypeSolver.class.getCanonicalName(), new ArrayList<String>());
```
Comment (xingweitian) [inline tests/checkers/inference/NninfTest.java:22]: It seems like the type arguments can be inferred. Can we try to remove them?

File: src/nninf/qual/NonNull.java (hunk 1 above)
Comment (zcai1) [reply src/nninf/qual/NonNull.java]: Thanks for pointing that out!

File: tests/checkers/inference/NninfTest.java (hunk 6 above)
Comment (zcai1) [reply tests/checkers/inference/NninfTest.java]: I think it's fine to remove it. I saw this comment in other subclasses of `CFInferenceTest`, so I thought it was a pattern we follow...

File: tests/checkers/inference/NninfTest.java (hunk 7 above)
Comment (zcai1) [reply tests/checkers/inference/NninfTest.java]: Yes, we should remove them. I'm not sure why we have these type arguments in other subclasses.

**Comment by zcai1** [comment]
@wmdietl @xingweitian 
Thank you for your feedback! I think this PR is ready for a final review, but we can only merge it after https://github.com/typetools/checker-framework/pull/3301 is merged and pulled to opprop.

**Comment by zcai1** [comment]
For the following test case:
```
class SimpleNninfTest1 {
    String id = "12345";
    Integer count = 0;
}
```

The nninf generates:
```
import nninf.qual.Nullable;
@Nullable class SimpleNninfTest1 {
    String id = "12345";
    @Nullable Integer count = ((@Nullable int) (0));
}
```
which obviously can't pass the final type check (literal int 0 should be non-null).

**Comment by zcai1** [comment]
The commit above makes `@Nonnull` the qualifier for primitive literals, which seems reasonable to me, but is different from the definition of `@Nonnull` in CF. 

Now nninf generates:
```
import nninf.qual.Nullable;
@Nullable class SimpleNninfTest1 {
    String id = "12345";
    Integer count = 0;
}
```

I think the `@Nullable` annotated class is another bug we have.

**Pull request merged**